pkgs/pkgs-lib/formats/hocon/test/default.nix

```
@@ -0,0 +1,5 @@
+{ pkgs, ... }:
+{
+  comprehensive = pkgs.callPackage ./comprehensive { };
+  backwards-compatibility = pkgs.callPackage ./backwards-compatibility { };
```

Here's a perhaps cleaner alternative to `_disableDeprecationWarnings`:

```suggestion
  backwards-compatibility =
    let
      pkgsNoWarn = pkgs.extend (final: prev: {
        lib = prev.lib.extend (libFinal: libPrev: {
          trivial = libPrev.trivial // {
            warn = msg: v: v;
          };
        });
      });
    in pkgsNoWarn.callPackage ./backwards-compatibility { };
```

---

This doesn't work, and I'm having a hard time figuring out why. Could it be that the `lib` used for `callPackage` is already captured in a closure somewhere? (ftr, I've also tried directly overriding `lib.warn` without going through `lib.trivial`, but it yields the same result)

---

Pushed the broken changes as a fixup commit